### PR TITLE
Use default config and partial overrides

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes

Refactoring the way Snakemake configuration is handled in this workflow to be more consistent with the ncov workflow.

### Related issue(s)

- Prompted by https://github.com/nextstrain/docs.nextstrain.org/pull/164#discussion_r1294800635

### Testing

- [x] Checks pass

### CODE REVIEW

File: config/defaults.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-# any values in this file.
+# This configuration file contains all required configuration entries with
+# hmpxv1-focused default values.
+# Use Snakemake's --configfile/--config options to override default values.
 
 exclude: "config/exclude_accessions_mpxv.txt"
 reference: "config/reference.fasta"
 genemap: "config/genemap.gff"
 genbank_reference: "config/reference.gb"
+colors: "config/colors_hmpxv1.tsv"
```
Comment (joverlee521) [inline config/defaults.yaml:9]: _nitpick_

I can't find where this config is being accessed in the workflow.
Maybe this should have been removed as part of https://github.com/nextstrain/monkeypox/pull/109?

**Review by joverlee521 — APPROVED** [review]
Thanks for making this change! Having a reasonable default config file is definitely a change I want to push across all Nextstrain pathogen repos!

File: config/defaults.yaml (hunk 1 above)
Comment (victorlin) [reply config/defaults.yaml:9]: Hmm, I don't see any references either. I'll make another PR to remove it.

Comment (victorlin) [reply config/defaults.yaml:9]: #174

**Comment by corneliusroemer** [comment]
Hmm, I don't like having an extra default config file which makes it more difficult to trace where a particular config value comes from.

PR #171 reworks things a little anyways. I'll add a default profile which specifies a default config file so things can be run as `snakemake`.

**Pull request merged**